### DISCUSSION TITLE
PDF draw rectangle  only page0 reversed

### DISCUSSION THREAD

**Discussion by HelloKitty2022** [pymupdf/PyMuPDF#912] [Looking for help] [answered]
doc=fitz.open(fname)
for i in range(len(doc)):
    page=doc[i]
    blocks=page.get_text("blocks")
    dicts=page.get_text("dict")
    contents=doc.get_page_xobjects(i)
    lastblock=[]
    for j in range(len(blocks)):

        block=blocks[j]
        text=block[4]
        if block[6]==0:
            textrect=(block[0]-0.5,block[1]-0.5,block[2]+0.5,block[3]+0.5)
            page.draw_rect(rect=textrect,color=(1,0,0),width=0.5)
            page.insert_text(textrect[0:2],str(j)+": ")
doc.save(fname+"done.pdf")

**Comment by HelloKitty2022** [comment]
[123.pdf](https://github.com/pymupdf/PyMuPDF/files/6010240/123.pdf)

**Comment by HelloKitty2022** [comment]
![Snipaste_2021-02-19_21-21-23](https://user-images.githubusercontent.com/30066186/108509796-ab84e000-72f8-11eb-941a-1e6e30685722.jpg)
![111](https://user-images.githubusercontent.com/30066186/108510298-4a114100-72f9-11eb-8348-063cc54eaaa6.png)

**Comment by HelloKitty2022** [comment]
other pages draw rect normal works 
in the same time  i want merge the blocks to a paragraph，if there is any simple other guys have done this
thanks

**Comment by JorjMcKie** [comment, marked as answer]
the page has what I call a "global geometry change" - please read the documentation here [Misplaced Item Insertions on PDF Pages](https://pymupdf.readthedocs.io/en/latest/faq.html#common-issues-and-their-solutions).
Detect this by checking `page.is_wrapped()`, if false execute `page.wrap_contents()`.
